tests/lib/managers/terms-of-service-test.js

```diff
@@ -0,0 +1,571 @@
+			return termsOfService.updateUserStatuses(TERMS_OF_SERVICE_USER_STATUS_ID, true)
+				.then(termsOfServiceUserStatusObject => {
+					assert.strictEqual(termsOfServiceUserStatusObject, response, 'terms of service user status object is returned');
+				});
+		});
+	});
+
+	describe('setUserStatuses()', function() {
+		var expectedParam = {body: {tos: {tos_id: '1234', type: 'terms_of_service'}, is_accepted: true}};
+		it('should make POST request to create custom terms of service user status when called', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service_user_status',
+					id: '1234',
+					tos: {},
+					user: {},
+					is_accepted: true,
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.mock(boxClientFake).expects('post').withArgs('/terms_of_service_user_statuses', expectedParam)
+				.returns(Promise.resolve(response));
+			termsOfService.setUserStatuses(TERMS_OF_SERVICE_ID, true, TERMS_OF_SERVICE_USER_ID);
+		});
+
+		it('should call callback with the created terms of service user status info when a 200 response is returned', function(done) {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service_user_status',
+					id: '1234',
+					tos: {},
+					user: {},
+					is_accepted: true,
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.stub(boxClientFake, 'post').returns(Promise.resolve(response));
+			termsOfService.setUserStatuses(TERMS_OF_SERVICE_ID, true, {user_id: '5678'}, function(err, termsOfServiceUserStatusObject) {
+				assert.ifError(err);
+				assert.strictEqual(termsOfServiceUserStatusObject, response.body, 'terms of service user object is returned');
+				done();
+			});
+		});
+
+		it('should return a promise resolving to created terms of service user status object when a 200 response is returned', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service_user_status',
+					id: '1234',
+					tos: {},
+					user: {},
+					is_accepted: true,
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.stub(boxClientFake, 'post').returns(Promise.resolve(response));
+			return termsOfService.setUserStatuses(TERMS_OF_SERVICE_ID, true, TERMS_OF_SERVICE_USER_ID)
+				.then(termsOfServiceUserStatusObject => {
+					assert.strictEqual(termsOfServiceUserStatusObject, response.body, 'terms of service user status object is returned');
+				});
+		});
+
+		it('should make PUT request to retrieve custom terms of service user status when called', function() {
```

Comment:
This test name should be updated to include the circumstances when the PUT call would be made, e.g. "should make PUT request to update user status when user status already exists"